execution/executor/src/components/in_memory_state_calculator_v2.rs

```diff
@@ -104,30 +106,46 @@ impl InMemoryStateCalculatorV2 {
 
         let sharded_state_cache = arr![DashMap::new(); 16];
         // TODO(grao): Move this logic to an earlier stage.
-        let _: Vec<_> = state_cache
-            .into_iter()
-            .map(|(k, v)| {
-                sharded_state_cache[k.get_shard_id() as usize].insert(k, v);
-            })
-            .collect();
+        let _: Vec<_> = {
+            let _timer = APTOS_EXECUTOR_OTHER_TIMERS_SECONDS
+                .with_label_values(&["produce_sharded_state_cache"])
+                .start_timer();
+            state_cache
+                .into_iter()
+                .map(|(k, v)| {
```

Comment:
`for_each()` and you don't need to collect

Reply:
Done